sound/soc/sof/intel/hda-loader.c

```diff
@@ -382,6 +383,64 @@ static int hda_dsp_boot_imr(struct snd_sof_dev *sdev)
 	return ret;
 }
 
+int hda_dsp_ipc4_load_library(struct snd_sof_dev *sdev,
+			      struct snd_sof_module_library_info *lib_info)
+{
+	struct hdac_ext_stream *hext_stream;
+	struct firmware stripped_firmware;
+	struct snd_dma_buffer dmab;
+	struct sof_ipc4_msg msg = {};
+	int ret, ret1;
+
+	stripped_firmware.data = lib_info->fw->data + lib_info->fw_offset;
+	stripped_firmware.size = lib_info->fw->size - lib_info->fw_offset;
```

Comment:
maybe check that the offset is smaller than a size?

Reply:
the parse_ext_manifest op already takes of it isnt it?

Reply:
no idea, a comment wouldn't hurt :-)